server/testdb/data_stores.go

```diff
@@ -223,12 +225,36 @@ func (td *postgresDB) readDataStoreRow(ctx context.Context, row scanner) (model.
 			return model.DataStore{}, fmt.Errorf("cannot parse data store: %w", err)
 		}
 
-		return dataStore, nil
+		return patchDataStoreValues(dataStore), nil
 	default:
 		return model.DataStore{}, err
 	}
 }
 
+func patchDataStoreValues(dataStore model.DataStore) model.DataStore {
+	if dataStore.Values.ElasticApm == nil {
```

Comment:
When adding the elastic APM object to the datastore values, it was causing an issue because the object was empty and when trying to run a deep copy it would fail, this check would patch any further incompatibility issues that we might encounter.